packages/preview-middleware-client/src/adp/quick-actions/fe-v4/enable-variant-management.ts

```diff
@@ -0,0 +1,86 @@
+import type FlexCommand from 'sap/ui/rta/command/FlexCommand';
+
+import { QuickActionContext, SimpleQuickActionDefinition } from '../../../cpe/quick-actions/quick-action-definition';
+
+import { SimpleQuickActionDefinitionBase } from '../simple-quick-action-base';
+import Component from 'sap/ui/core/Component';
+import { getUi5Version, isLowerThanMinimalUi5Version } from '../../../utils/version';
+import { createManifestPropertyChange } from '../../../utils/fe-v4';
+import ListReportComponent from 'sap/suite/ui/generic/template/ListReport';
+
+export const ENABLE_VARIANT_MANAGEMENT_IN_TABLES_CHARTS = 'enable-variant-management-in-tables-charts';
+
+// sap.f.DynamicPage for list report and sap.uxap.ObjectPageLayout for object page.
+const CONTROL_TYPES = ['sap.f.DynamicPage', 'sap.uxap.ObjectPageLayout'];
+
+/**
+ * Quick Action for enabling table filtering using table personalization settings.
+ */
+export class EnableVariantManagementQuickAction
+    extends SimpleQuickActionDefinitionBase
+    implements SimpleQuickActionDefinition
+{
+    private pageSmartVariantManagementMode = '';
+    private ownerComponent: ListReportComponent;
```

Comment:
Looking at the code on line 62-64, it could be either of the three different components. The type here doesn't reflect that.

Reply:
Decided to fix in in the separate pr